sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -85,10 +88,20 @@ case class Generate(
 
     if (join) child.output ++ qualified else qualified
   }
+
+  override protected def otherJsonValues: JValue = {
+    ("generator-type" -> generator.getClass.getSimpleName) ~
+      ("join" -> join) ~
+      ("outer" -> outer) ~
+      ("qualifier" -> qualifier) ~
+      ("generator-output" -> generatorOutput.map(_.simpleString))
```

Comment:
Since expressions are trees too, it would be nice if we could serialize them also.  We are loosing lots of information about types by saving the `simpleString` only.